### ISSUE TITLE
Date Picker - type range

### ISSUE THREAD

**Issue by gabisinhas** [carbon-design-system/carbon-components-vue#1625]
Date Picker type range is not returning event output with start date and end date. It is returning just start date.
![image](https://github.com/user-attachments/assets/043b495f-8c93-4d79-9d1b-5b25082e54a8)

## Detailed description

> Describe in detail the issue you're having.

> Is this a feature request (new component, new icon), a bug, or a general issue?
- bug

> Is this issue related to a specific component?
- Date Picker type range

> What version of the Carbon Design System are you using?
    "carbon-components": "^10.58.15",

## Additional information

- Screenshots or code
- Notes

**Comment by davidnixon** [comment]
I will update the range version to emit the change event as an object similar to how the v-model works. 
I'll double check the react component to see how that version emits the change events.

**Comment by davidnixon** [comment]
The onChange in Carbon react emits an array with 2 date objects. i.e.
```js
[Wed Sep 04 2024 00:00:00 GMT-0400 (Eastern Daylight Time), Thu Sep 19 2024 00:00:00 GMT-0400 (Eastern Daylight Time)]
```

The fix for this should follow that pattern.

**Comment by davidnixon** [comment]
For `kind="single"` Carbon React issues an array of 1 date object. I was surprised by this as a array with a single object is not very intuitive but looking at the code, this event is coming directly from the underlying flat picker component. The fix for this issue should follow this same pattern of an array with single date object.

**Comment by davidnixon** [comment]
fix in 3.0.21
